### PR TITLE
Euth allow private processes

### PR DESCRIPTION
**Pull request by pallix**
This PR breaks one workflow test. @joka proposed to fix it.

### CODE REVIEW

**Comment by pallix** [comment]
Concept is working, I need to adjust all permissions

**Comment by pallix** [comment]
Ok I think I'm done with this PR. @joka review :-) ?

**Comment by pallix** [comment]
Ideally I would prefer to have the tests of the private workflows in core but for the moment it's not so easy since the document_process and core process use the 'sample' workflow instead of 'debate' / 'standard'.

**Comment by pallix** [comment]
@xi @MagdaN private processes have a different type: `IPrivateProcess` in the same module as the other. JSON files are in the `workflows` folder.

**Comment by xi** [comment]
> private processes have a different type: `IPrivateProcess`

That is an issue. The frontend needs a way to determine whether a resource is a process or not. Can you maybe add a common supertype?

File: src/adhocracy_core/adhocracy_core/testing.py [hunk 1]
```diff
@@ -687,9 +699,10 @@ def make_configurator(app_settings: dict, package) -> Configurator:
...
     # in functional tests.
 
-    def set_acm_subscriber(event):
-        set_acms_for_app_root(event.app, (root_acm,))
-    configurator.add_subscriber(set_acm_subscriber, ApplicationCreated)
+    # FIXME  this creates a problem by euth, since euth override the root acl
+    # def set_acm_subscriber(event):
+    #     set_acms_for_app_root(event.app, (root_acm,))
+    # configurator.add_subscriber(set_acm_subscriber, ApplicationCreated)
     return configurator
```
Comment (joka) [inline src/adhocracy_core/adhocracy_core/testing.py]: TODO for joka

File: src/adhocracy_euth/adhocracy_euth/conftest.py [hunk 2]
```diff
@@ -2,6 +2,15 @@
 from pytest import fixture
 
 
+@fixture(scope='session')
+def app_settings(app_settings) -> dict:
+    """Return settings to start the test wsgi app."""
+    # disable creating a default group, this causes
+    # ZODB.POSException.InvalidObjectReference
+    app_settings['adhocracy.add_default_group'] = True
```
Comment (joka) [inline src/adhocracy_euth/adhocracy_euth/conftest.py]: comment is needless

File: src/adhocracy_euth/adhocracy_euth/workflows/test_public_private_process.py [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+    def test_create_resources(self,
+                               # FIXME: avoid triggering error during commit
+                              registry,
+                              datadir,
+                              process_url_private,
+                              process_url_public,
+                              app_admin):
+        # import sys
+        # sys.debug = False
+        root = get_root(app_admin.app_router)
```
Comment (joka) [inline src/adhocracy_euth/adhocracy_euth/workflows/test_public_private_process.py]: needless comments

File: src/adhocracy_core/adhocracy_core/workflows/standard.py [hunk 4]
```diff
@@ -10,53 +10,64 @@
...
-                                  [['create_proposal',            'Allow',        None,        None,      None],
-                                   ['edit_proposal',               None,          None,       'Allow',    None],
-                                   ['create_comment',             'Allow',       'Allow',      None,      None],
-                                   ['edit_comment',                None,          None,       'Allow',    None],
-                                   ['create_rate',                'Allow',        None,        None,      None],
-                                   ['edit_rate',                   None,          None,       'Allow',    None],
-                                   ['create_relation',            'Allow',        None,        None,      None],
-                                   ['edit_relation',               None,          None,       'Allow',    None]
+                                  [['view',                       None,       'Allow',       'Allow',     'Allow',   'Allow',      None],
+                                   ['create_proposal',            None,       'Allow',        None,        None,      None,        None],
```
Comment (joka) [inline src/adhocracy_core/adhocracy_core/workflows/standard.py]: TODO joka update view row in workflow spreadsheet

File: src/adhocracy_euth/adhocracy_euth/resources/collaborative_text.py [hunk 5]
```diff
@@ -13,6 +13,17 @@ class IProcess(document_process.IDocumentProcess):
 )
 
 
+class IPrivateProcess(document_process.IDocumentProcess):
+    """Private collaborative text participation process."""
+
```
Comment (joka) [inline src/adhocracy_euth/adhocracy_euth/resources/collaborative_text.py]: IPrivateProcess should subclass IProcess, it is actually the same process type exept different permisions.

File: src/adhocracy_euth/adhocracy_euth/resources/idea_collection.py [hunk 6]
```diff
@@ -15,6 +15,18 @@ class IProcess(process.IProcess):
 )
 
 
+class IPrivateProcess(process.IProcess):
+    """Private idea collection participation process."""
+
```
Comment (joka) [inline src/adhocracy_euth/adhocracy_euth/resources/idea_collection.py]: .IPrivateProcess should subclass .IProcess, it is actually the same process type exept different permisions.

File: src/adhocracy_euth/adhocracy_euth/workflows/test_idea_collection.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
+from pytest import mark
```
Comment (joka) [inline src/adhocracy_euth/adhocracy_euth/workflows/test_idea_collection.py]: The tests for adhocracy_core workflow should be in adhocracy_core

**Comment by pallix** [comment]
@joka my last changes break bplane tests

**Comment by joka** [comment]
The authentication.rst doctest is still failing, but we should merge this now to make the release today.
I'll fix this soon.

**Comment by joka** [comment]
I fixed the setup for functional workflow tests (https://github.com/liqd/adhocracy3/pull/2075). There should be no sideeffects if you use the helper methods in adhocracy_core.utils.testing.

**Comment by joka** [comment]
All tests are fixed now

**Pull request merged**